libnmstate/ifaces/macvlan.py

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.error import NmstateValueError
+from libnmstate.schema import MacVlan
+
+from .base_iface import BaseIface
+
+
+class MacVlanIface(BaseIface):
+    @property
+    def parent(self):
+        return self._macvlan_config.get(MacVlan.BASE_IFACE)
+
+    @property
+    def need_parent(self):
+        return True
+
+    @property
+    def _macvlan_config(self):
+        return self.raw.get(MacVlan.CONFIG_SUBTREE, {})
+
+    @property
+    def is_virtual(self):
+        return True
+
+    @property
+    def can_have_ip_when_enslaved(self):
+        return True
```

Comment:
MAC VLAN cannot hold IP address when become part of bridge/bond/team.

Actually the code VLAN/VXLAN is also wrong on this. Can you create a PR fixing that also?

Reply:
Sure, thanks!